libs/ui-components/src/components/Device/DeviceDetails/DeviceDetailsPage.tsx

```diff
@@ -29,6 +31,9 @@ const DeviceDetailsPage = ({ children, hideTerminal }: React.PropsWithChildren<{
 
   const deviceAlias = device?.metadata.labels?.alias || deviceId;
 
+  const [canDelete] = useAccessReview(RESOURCE.DEVICE, VERB.DELETE);
+  const [canEdit] = useAccessReview(RESOURCE.DEVICE, VERB.PATCH);
```

Comment:
There should also be a check in `EditDeviceWizard` to prevent the users from accessing the "edit" URL directly.
Can be done as a follow-up.

Reply:
the same would apply for any other URL that can be accessed directly (edit fleet, repo etc.).
I can fix that in a followup.

Reply:
The follow up is tracked in https://issues.redhat.com/browse/EDM-949